streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -232,12 +236,14 @@ private void updateHead() {
             }
             headRecord = new StampedRecord(deserialized, timestamp);
             headRecordSizeInBytes = consumerRecordSizeInBytes(raw);
+            rawHeadRecord = raw;
```

Comment:
We considered putting the rawRecord into the StampedRecord, but this would have required changes to some signature methods.
This is why we chose to do it this way.

Reply:
What do you mean? `StampedRecord` is not public API. You can change it w/o a KIP.

Reply:
Adding a new attribute to `StampedRecord` is not that impactful we thought. We can definitely do it

This only impacts [CorruptedRecord class](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/CorruptedRecord.java), where `record` will be equal to `rawRecord`

Reply:
Done.

Raw record has been added to `StampedRecord`.